drivers/usb_c/tcpc/ucpd_numaker.c

```diff
@@ -0,0 +1,2558 @@
+	uint32_t ccsts_cc2state = ccsts & UTCPD_CCSTS_CC2STATE_Msk;
+	uint32_t ccsts_conrlt = ccsts & UTCPD_CCSTS_CONRLT_Msk;
+
+	/* CC1 */
+	if (rolctl_cc1 == UTCPD_ROLECTL_CC1_RP || ccsts_conrlt == UTCPD_CONN_RESULT_RP) {
+		switch (ccsts_cc1state) {
+		case UTCPD_CCSTS_CC1STATE_SRC_RA:
+			*cc1 = TC_CC_VOLT_RA;
+			break;
+		case UTCPD_CCSTS_CC1STATE_SRC_RD:
+			*cc1 = TC_CC_VOLT_RD;
+			break;
+		default:
+			*cc1 = TC_CC_VOLT_OPEN;
+		}
+	} else if (rolctl_cc1 == UTCPD_ROLECTL_CC1_RD || ccsts_conrlt == UTCPD_CONN_RESULT_RD) {
+		switch (ccsts_cc1state) {
+		case UTCPD_CCSTS_CC1STATE_SNK_DEF:
+			*cc1 = TC_CC_VOLT_RP_DEF;
+			break;
+		case UTCPD_CCSTS_CC1STATE_SNK_1P5A:
+			*cc1 = TC_CC_VOLT_RP_1A5;
+			break;
+		case UTCPD_CCSTS_CC1STATE_SNK_3A:
+			*cc1 = TC_CC_VOLT_RP_3A0;
+			break;
+		default:
+			*cc1 = TC_CC_VOLT_OPEN;
+		}
+	} else {
+		*cc1 = TC_CC_VOLT_OPEN;
+	}
+
+	/* CC2 */
+	if (rolctl_cc2 == UTCPD_ROLECTL_CC2_RP || ccsts_conrlt == UTCPD_CONN_RESULT_RP) {
+		switch (ccsts_cc2state) {
+		case UTCPD_CCSTS_CC2STATE_SRC_RA:
+			*cc2 = TC_CC_VOLT_RA;
+			break;
+		case UTCPD_CCSTS_CC2STATE_SRC_RD:
+			*cc2 = TC_CC_VOLT_RD;
+			break;
+		default:
+			*cc2 = TC_CC_VOLT_OPEN;
+		}
+	} else if (rolctl_cc2 == UTCPD_ROLECTL_CC2_RD || ccsts_conrlt == UTCPD_CONN_RESULT_RD) {
+		switch (ccsts_cc2state) {
+		case UTCPD_CCSTS_CC2STATE_SNK_DEF:
+			*cc2 = TC_CC_VOLT_RP_DEF;
+			break;
+		case UTCPD_CCSTS_CC2STATE_SNK_1P5A:
+			*cc2 = TC_CC_VOLT_RP_1A5;
+			break;
+		case UTCPD_CCSTS_CC2STATE_SNK_3A:
+			*cc2 = TC_CC_VOLT_RP_3A0;
+			break;
+		default:
+			*cc2 = TC_CC_VOLT_OPEN;
+		}
+	} else {
+		*cc2 = TC_CC_VOLT_OPEN;
+	}
+
+	return 0;
+}
+
+/**
+ * @brief Sets the value of CC pull up resistor used when operating as a Source
+ *
+ * @retval 0 on success
+ * @retval -EIO on failure
+ */
+static int numaker_tcpc_select_rp_value(const struct device *dev, enum tc_rp_value rp)
+{
+	struct numaker_tcpc_data *data = dev->data;
+
+	data->rp = rp;
+
+	return 0;
+}
+
+/**
+ * @brief Gets the value of the CC pull up resistor used when operating as a Source
+ *
+ * @retval 0 on success
+ * @retval -EIO on failure
+ */
+static int numaker_tcpc_get_rp_value(const struct device *dev, enum tc_rp_value *rp)
+{
+	struct numaker_tcpc_data *data = dev->data;
+
+	*rp = data->rp;
+
+	return 0;
+}
+
+/**
+ * @brief Sets the CC pull resistor and sets the role as either Source or Sink
+ *
+ * @retval 0 on success
+ * @retval -EIO on failure
+ */
+static int numaker_tcpc_set_cc(const struct device *dev, enum tc_cc_pull pull)
+{
+	const struct numaker_tcpc_config *const config = dev->config;
+	struct numaker_tcpc_data *data = dev->data;
+	UTCPD_T *utcpd_base = config->utcpd_base;
+	int rc;
+	uint32_t rolctl = 0;
+
+	/* Disable Dead Battery mode if it is active, so that
+	 * internal Rd/Rp gets controlled by to UTCPD.ROLCTL
+	 * from Dead Battery circuit.
+	 */
+	rc = numaker_utcpd_deadbattery_query_enable(dev);
+	if (rc < 0) {
+		return rc;
+	}
```

Comment:
this cannot happen, `numaker_utcpd_deadbattery_query_enable` returns either 0 or 1. Maybe make `numaker_utcpd_deadbattery_query_enable` a bool and then handle it accordingly?

Reply:
changed return type to `bool` and made related modification accordingly